### PR TITLE
[api server] fix job server problems

### PR DESCRIPTION
**Pull request by Basasuya**
## Why are these changes needed?

* fix description error in job.proto
* tune MarshalOptions in http proxy
* fix job trasfer function

## Checks

- [*] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [*] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Basasuya** [comment]
@Jeffwan cc

File: apiserver/cmd/main.go [hunk 1]
```diff
@@ -85,8 +85,9 @@ func startHttpProxy() {
 	runtimeMux := runtime.NewServeMux(
 		runtime.WithMarshalerOption(runtime.MIMEWildcard, &runtime.JSONPb{
 			MarshalOptions: protojson.MarshalOptions{
-				UseProtoNames:  true,
+				UseProtoNames:  false,
```
Comment (Jeffwan) [inline apiserver/cmd/main.go:88]: need some explanation here and `allowPartial`. Do you meet some problem on the `shutdown_after_job_finishes ` or `shutdownAfterJobFinishes `

File: apiserver/pkg/model/converter.go [hunk 2]
```diff
@@ -156,12 +146,19 @@ func FromCrdToApiJob(job *v1alpha1.RayJob) (pbJob *api.RayJob) {
...
 		ClusterSpec:              PopulateRayClusterSpec(job.Spec.RayClusterSpec),
-		TtlSecondsAfterFinished:  ttl,
 		CreatedAt:                &timestamp.Timestamp{Seconds: job.CreationTimestamp.Unix()},
-		DeleteAt:                 &timestamp.Timestamp{Seconds: deleteTime},
 		JobStatus:                string(job.Status.JobStatus),
 		JobDeploymentStatus:      string(job.Status.JobDeploymentStatus),
 		Message:                  job.Status.Message,
 	}
+
+	if job.Spec.TTLSecondsAfterFinished != nil {
```
Comment (Jeffwan) [inline apiserver/pkg/model/converter.go:155]: Just like to confirm that this is just improvement, there's no bug or nil issue, right?

Comment (Basasuya) [reply apiserver/pkg/model/converter.go:155]: yes

File: apiserver/cmd/main.go (hunk 1 above)
Comment (Basasuya) [reply apiserver/cmd/main.go:88]: java only accept transfer json string to structure with Camel-Case, so i set the useProtoNames;
allowPartial is useless, we have no required/optional label in proto, removed;
add DiscardUnknown in UnmarshalOptions to ensure universality for the input json.

Comment (Jeffwan) [reply apiserver/cmd/main.go:88]: So it's used by ORM mapping?

**Review by Jeffwan — APPROVED** [review]

File: apiserver/cmd/main.go (hunk 1 above)
Comment (Basasuya) [reply apiserver/cmd/main.go:88]: yes

**Pull request merged**